app/controllers/idv/gpo_only_warning_controller.rb

```diff
@@ -6,7 +6,7 @@ class GpoOnlyWarningController < ApplicationController
     before_action :confirm_two_factor_authenticated
 
     def show
-      user_session['idv/doc_auth'][:skip_vendor_outage] = true
+      user_session.fetch('idv/doc_auth', {})[:skip_vendor_outage] = true
```

Comment:
This is interesting. I hadn't seen this pattern before and I'm curious if you know why we used this flag in the session instead of `skip_before_action` for these actions we want to skip the vendor check?

Part of the reason I ask is it looks like this is going to be sticky. Once the user visits a controller that sets `skip_vendor_outage` then the vendor outage feature is essentially disabled for their session.

Reply:
Ah, I see now. Thanks!

> I was also wondering if I should add an analytics event for the warning page being visited

Yes, we should add an event for sure

> And see queries in the PR description about personal_key, etc.

I don't see any questions in the PR description but maybe I am looking in the wrong spot.

Reply:
> I don't see any questions in the PR description but maybe I am looking in the wrong spot.
Oh, that was in #8643 that I just merged:
personal_key_controller, gpo_controller and gpo_verify_controller don't include IdvStepConcern. Should they? They also don't include check_for_outage and handle_fraud before actions.

Reply:
I think it makes sense that those later controllers do not include these checks, because there are no external services used by that point that could potentially experience an outage, right?